### PR TITLE
[Speculative decoding] CUDA graph support

### PR DESCRIPTION
**Pull request by heeju-kim2**
We add a end-to-end test that verifies speculative decoding works with CUDA graphs. Specifically, we ensure that a model produces the same output when speculative decoding is enabled vs not, when cuda graphs are enabled.

### CODE REVIEW

**Comment by cadedaniel** [comment]
I will take a look today

File: examples/offline_inference.py [hunk 1]
```diff
@@ -19,4 +19,7 @@
 for output in outputs:
     prompt = output.prompt
     generated_text = output.outputs[0].text
-    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
+    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}"
+
+
+
```
Comment (mgoin) [inline examples/offline_inference.py:25]: These look like mistaken changes/invalid syntax - please restore this file

**Comment by cadedaniel** [comment]
Thanks again @heeju-kim2 for the PR.

So I ran spec decode locally with cuda graphs enabled, and I find that it works. Are you misconfiguring spec decode?

You can see what I saw by running the test below, which uses [`run_greedy_equality_correctness_test` from test_correctness.py](https://github.com/vllm-project/vllm/blob/1e8f4252aa163041094a8fedbb701a98d7087d7c/tests/spec_decode/e2e/test_correctness.py#L543).

By the way, if this works for you, you can update this PR to simply include this test, remove the other changes, and we can merge it. Thanks for contributing to vLLM :)

```python3
@pytest.mark.parametrize(
    "common_llm_kwargs",
    [{  
        # Required for spec decode.
        "use_v2_block_manager": True,
    }]) 
@pytest.mark.parametrize("per_test_common_llm_kwargs", [
{
    # Identical models.
    "model": "JackFram/llama-68m",
    "speculative_model": "JackFram/llama-68m",
    "num_speculative_tokens": 5,
},
{
    # Distinct models.
    "model": "JackFram/llama-160m",
    "speculative_model": "JackFram/llama-68m",
    "num_speculative_tokens": 5,
}
])
@pytest.mark.parametrize("baseline_llm_kwargs", [{}])
@pytest.mark.parametrize("test_llm_kwargs", [{
    "enforce_eager": False,
}])
@pytest.mark.parametrize("batch_size", [1, 32])
@pytest.mark.parametrize("output_len", [128])
@pytest.mark.parametrize("seed", [1])
def test_spec_decode_cuda_graph(baseline_llm_generator, test_llm_generator, batch_size, output_len):
    run_greedy_equality_correctness_test(
        baseline_llm_generator,
        test_llm_generator,
        batch_size,
        max_output_len=output_len,
        force_output_len=True,
    )
```

**Comment by cadedaniel** [comment]
Thanks for the PR @heeju-kim2 !

**Review by cadedaniel — APPROVED** [review]

**Pull request merged**